implementations/rust/ockam/ockam_api/src/kafka/protocol_aware.rs

```diff
@@ -59,16 +95,120 @@ impl ProtocolState {
         };
 
         if let Ok(api_key) = ApiKey::try_from(header.request_api_key) {
-            info!(
-                "request: length: {}, version {:?}, api {:?}",
+            trace!(
+                "request: length: {}, correlation {}, version {}, api {:?}",
                 buffer.len(),
+                header.correlation_id,
                 header.request_api_version,
                 api_key
             );
 
-            #[allow(clippy::single_match)]
             match api_key {
-                ApiKey::MetadataKey => {
+                ApiKey::ProduceKey => {
```

Comment:
This is a large function. It could be a good idea to just see the top-level `match api_key` with the different cases and then delegate to specific functions (same thing for the response side).